me.mitya57.ReText.json

```diff
@@ -0,0 +1,378 @@
+                    "type": "shell",
+                    "commands": [
+                        "echo 'all:\n\ninstall:\n\tpip3 install --prefix=/app Markups-*.whl;\n' > makefile;"
+                    ]
+                }
+            ]
+        },
+        {
+            "name": "retext",
+            "no-autogen": true,
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://pypi.python.org/packages/2a/ba/581591a8a8d48d02236b57633ae5f55247424885e143fe1d21fba904a1ed/ReText-7.0.1.tar.gz",
+                    "sha256": "f5a2dbfb32b96423fd3e5ef00488440c36a2bcecbc6356f5c17674b265316eea"
+                },
+                {
+                    "type": "archive",
+                    "url": "https://github.com/mquinson/po4a/archive/v0.52.tar.gz",
+                    "sha256": "547c410ad3bb8e5fd3939a6acdd39713529187be05fca7235095d57d3f0c6cfc",
+                    "strip-components": 0
+                },
+                {
+                    "type": "file",
+                    "path": "retext-man-de.po",
+                    "sha256": "9b00b30a693a3023b2f5619ab0586ddf9bd8f5bc9b75c2dabd5d6493558bb6f7"
+                },
+                {
+                    "type": "file",
+                    "path": "retext.1",
+                    "sha256": "300f04c1f808f63351b16ed0fa0f2d01b857c4d096ed71dcf54ab95281727d9a"
+                },
+                {
+                    "type": "shell",
+                    "commands": [
+                        "echo 'disabling external renderer';",
+                        "cp 'ReText/window.py' 'ReText/window.py.bak';",
+                        "cp 'ReText/tab.py' 'ReText/tab.py.bak';",
+                        "sed -r -i 's/^([[:blank:]]*)(menuEdit\\.addAction\\(self\\.(actionWebKit|actionWebEngine)\\))([[:blank:]]*)$/\\1pass #\\2\\4/' 'ReText/window.py';",
+                        "sed -r -i 's/^([[:blank:]]*)(globalSettings\\.(useWebKit|useWebEngine)[[:blank:]]*=[[:blank:]]*)(enable)([[:blank:]]*)$/\\1\\2False\\5/' 'ReText/window.py';",
+                        "sed -r -i 's/^(([[:blank:]]*)(if ReTextWebKitPreview and globalSettings\\.useWebKit:)([[:blank:]]*))$/\\2return ReTextPreview\\(self\\)\\n\\1/' 'ReText/tab.py';",
+                        "echo 'ReText/window.py';",
+                        "diff 'ReText/window.py.bak' 'ReText/window.py' -C 4 || :;",
+                        "echo 'ReText/tab.py';",
+                        "diff 'ReText/tab.py.bak' 'ReText/tab.py' -C 4 || :;",
+                        "rm -f 'ReText/window.py.bak' 'ReText/tab.py.bak';"
+                    ]
+                },
+                {
+                    "type": "shell",
+                    "commands": [
+                        "echo 'all:\n\ninstall:\n\tpython3 setup.py build; python3 setup.py install -O1 --skip-build --prefix=/app --root=/;\n' > makefile;"
+                    ]
+                }
+            ],
+            "post-install": [
```

Comment:
This is massive and complex enough that it should be its own file probably. It also should try to make it upstream.

Reply:
To be clear here again since there seems to be confusion. I mean putting it in a bash file outside of json so you can have sane whitespacing, no escaping of text, syntax highlighting, etc.